FirebaseSessions/Sources/SessionGenerator.swift

```diff
@@ -39,15 +39,16 @@ struct SessionInfo {
 ///
 class SessionGenerator {
   private var thisSession: SessionInfo?
-  private var settings: SettingsProtocol
 
   private var firstSessionId = ""
   private var sessionIndex: Int32
+  private var collectEvents: Bool
 
-  init(settings: SettingsProtocol) {
-    self.settings = settings
+  init(collectEvents: Bool) {
     // This will be incremented to 0 on the first generation
     sessionIndex = -1
+
+    self.collectEvents = collectEvents
```

Comment:
Where is this value used? I mean where do you check if collectEvents is true and so generate a new session or otherwise?

I'm also trying to validate the place where the config fetches happen irrespective of the state of data collection.

Reply:
This is used in constructing the SessionInfo object below: https://github.com/firebase/firebase-ios-sdk/pull/10769/files#diff-55a6b65c4c6ce0a2d723a584879ca1d8bfb4f621e449d935a34e73e6462c6defR67

Reply:
This is where config fetches happen: https://github.com/firebase/firebase-ios-sdk/blob/784c1012a22ac3a1e7eed0bbba652b8c3c6d3292/FirebaseSessions/Sources/FirebaseSessions.swift#L178

Reply:
Sounds good. Thank you!